docs/sql/01-query-optimizer.md

```diff
@@ -0,0 +1,689 @@
+# SQL Optimizer (state of 5.2.0)
+
+## Overview
+
+When a query string is submitted for execution, it is first parsed, converted to a tree of relational operators, and then
+passed to the optimizer in order to find the best execution path.
+
+In this document, we describe the design of the Hazelcast Mustang query optimizer that is based on Apache Calcite.
```

Comment:
`Mustang` name does not appear in the source code. Who is using it?

Reply:
It's the name of our optimizer. :)